pkg/tfshim/sdk-v2/internal/tf/plans/objchange/objchange.go

```diff
@@ -69,7 +69,10 @@ func PlannedDataResourceObject(schema *configschema.Block, config cty.Value) cty
 }
 
 func proposedNew(schema *configschema.Block, prior, config cty.Value) cty.Value {
-	if config.IsNull() || !config.IsKnown() {
+	if !config.IsKnown() {
```

Comment:
This file is marked with:

```
// Code copied from https://github.com/hashicorp/terraform.git by go generate; DO NOT EDIT.
```

Are you sure this change brings the overall system closer to TF Behavior? Can we place it outside this file to keep it vendored?

Reply:
Ah I see, I see, I'd guess this branch is never even hit in TF. So it's fine to edit. If you can place the edit at the call site of ObjChange not this file itself it can be easier to keep up with vendoring. Thank you!

Reply:
The function is recursive, so we can't do that easily. We could add a reconstruction step after this function which goes and recovers unknowns where they were discarded but I feel like this is better.

I'll see if there's a newer version of this upstream.

Reply:
Yup, updating the objchange copy did fix the issue.